assets/js/base/components/price-slider/index.js

```diff
@@ -26,321 +36,269 @@ const PriceSlider = ( {
 	showFilterButton,
 	isLoading,
 } ) => {
-	const [ value, setValue ] = useState( {
-		min: initialMin,
-		max: initialMax,
-	} );
-	const [ formattedValue, setFormattedValue ] = useState( {
-		min: formatCurrencyForInput( value.min, priceFormat, currencySymbol ),
-		max: formatCurrencyForInput( value.max, priceFormat, currencySymbol ),
-	} );
-	const debouncedChangeValue = useDebounce( value, 500 );
+	const minRange = useRef();
+	const maxRange = useRef();
+	const [ minPrice, setMinPrice ] = useState( initialMin );
+	const [ maxPrice, setMaxPrice ] = useState( initialMax );
+	const [ formattedMinPrice, setFormattedMinPrice ] = useState(
+		formatCurrencyForInput( minPrice, priceFormat, currencySymbol )
+	);
+	const [ formattedMaxPrice, setFormattedMaxPrice ] = useState(
+		formatCurrencyForInput( maxPrice, priceFormat, currencySymbol )
+	);
+	const debouncedChangeValue = useDebounce( [ minPrice, maxPrice ], 500 );
 
 	useEffect( () => {
 		if (
-			value.min === undefined ||
-			value.max === undefined ||
-			min > value.min ||
-			max < value.max
+			minPrice === undefined ||
+			maxPrice === undefined ||
+			minConstraint > minPrice ||
+			maxConstraint < maxPrice
 		) {
-			setValue( {
-				min,
-				max,
-			} );
+			setMinPrice( minConstraint );
+			setMaxPrice( maxConstraint );
 		}
-		setFormattedValue( {
-			min: formatCurrencyForInput(
-				value.min,
-				priceFormat,
-				currencySymbol
-			),
-			max: formatCurrencyForInput(
-				value.max,
-				priceFormat,
-				currencySymbol
-			),
-		} );
-	}, [ min, max ] );
+	}, [ minConstraint, maxConstraint ] );
```

Comment:
minPrice, and maxPrice should be dependencies as well otherwise this check won't run if they change.

Reply:
Doesn't need to; this check only needs to run if constraints change. If the min/max current changes it doesn't matter to this block - its within the constraints already.

Reply:
Hmm, what about changes in the input, or do we care about that?